File: Project1/fork.py
from multiprocessing import Process
import os
import time

def handle_fork():
    print("Process spawned \n")

def main():
    p = Process(target=handle_fork)
    p.start()
    p.join()

    print(f"Parent Process id: {os.getppid()}")
    print(f"Child Process id: {os.getpid()}")

    time.sleep(60)



if __name__ == "__main__":
    main()

File: Project1/fork.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <unistd.h>

int getProcessIds() {
    printf("Parent process id: %d\nChild process id: %d\n", getppid(), getpid());
    return 0;
};

int handleFork() {
    pid_t p;
    p = fork();

    if(p < 0) {
        printf("Error calling fork()");
        exit(1);
    }

    if(p == 0) {
        getProcessIds();
    }

    return 0;
};


int main() {
    handleFork();
    sleep(60);
    return 0;
};